src/Domain/UseCase/CreateValetToken/CreateValetTokenDto.ts

```diff
@@ -0,0 +1,15 @@
+import { Operation } from '../../Operation/Operation'
+import { Resource } from '../../Resource/Resource'
+import { UserWithPermissions } from '../../User/UserWithPermissions'
+import { ValidityPeriod } from '../../ValetToken/ValetToken'
+
+// from the API Gateway
+export type CreateValetKeyDto = {
+  // from auth
+  user: UserWithPermissions,
+  // requested by client
+  operation: Operation,
+  resources: Resource[],
+  // optional validity period
+  validityPeriod?: Partial<ValidityPeriod>,
```

Comment:
great idea 👍

Reply:
In this case I would view the comment `// optional validity period` as redundant indeed 😉